src/Util/Message.cc

```diff
@@ -447,9 +448,27 @@ CARTA::ChannelMapFlowControl Message::ChannelMapFlowControl(int32_t file_id, int
     return message;
 }
 
-CARTA::EventType Message::EventType(std::vector<char>& message) {
-    carta::EventHeader head = *reinterpret_cast<const carta::EventHeader*>(message.data());
-    return static_cast<CARTA::EventType>(head.type);
+carta::EventHeader Message::GetEventHeader(std::string_view message) {
+    return *reinterpret_cast<const carta::EventHeader*>(message.data());
+}
+
+/**
+ * @note This function creates a binary buffer containing a CARTA::EventHeader followed by the serialized protobuf message.
+ * The header includes the event type, protocol version, and event/request ID.
+ */
+std::vector<char> Message::EncodeMessage(CARTA::EventType event_type, uint32_t event_id, const google::protobuf::MessageLite& message) {
+    size_t message_length = message.ByteSizeLong();
+    size_t required_size = sizeof(carta::EventHeader) + message_length;
+
+    std::vector<char> msg(required_size, 0);
```

Comment:
I think the `0` parameter is unnecessary here -- I believe that this is the default value for initializing a vector with chars.